cdisc_rules_engine/services/define_xml/base_define_xml_reader.py

```diff
@@ -93,13 +93,25 @@ def extract_domain_metadata(self, domain_name: str = None) -> dict:
         return domain_metadata_dict
 
     @cached("define-variables-metadata")
-    def extract_variables_metadata(self, domain_name: str = None) -> List[dict]:
+    def extract_variables_metadata(
+        self, domain_name: str = None, **kwargs
+    ) -> List[dict]:
+        vx = kwargs.get("validate_xml", "Y")
         logger.info(
             f"Extracting variables metadata from Define-XML. domain_name={domain_name}"
         )
-        metadata = self._odm_loader.MetaDataVersion()
-        domain_metadata = self._get_domain_metadata(metadata, domain_name)
+        metadata = None
+        domain_metadata = None
         variables_metadata = []
+        try:
```

Comment:
1. I agree that it might need a new method to handle this. As to this validate-xml flag, it serves the purpose for now. 
2. The definition of validate-xml is to validate Define-XML or not when it misses def:Structure or in error:
      Y (skip define rule validation and continue for other rules)
       N (stop on error and report the error)

Reply:
It may serve it's purpose for this specific issue, but I don't think it is much more work to create a new method for the validation. I'm not saying you need to apply it in each function, but having it available will probably be useful in the future.

Reply:
Added get_metadata_version